ProjFS.Mac/PrjFSKextTests/MockVnodeAndMount.cpp

```diff
@@ -33,11 +38,68 @@ vnode::vnode(const shared_ptr<mount>& mount) :
 {
 }
 
+vnode::vnode(const std::shared_ptr<mount>& mount, VnodeCreationProperties properties) :
```

Comment:
We're pretty confident this will grow?  Looks like creation properties are only 3 parameters right now.

Reply:
I have a few issues with the existing approach:

 * There's not a good way to pick & mix the arguments you care about; default arguments only work right-to-left. The combinations I ran into were pretty arbitrary however.
 * With lots of arguments, it becomes non-obvious which one means what.
 * As you imply, it doesn't scale very well.

The C99 style named struct field initialisers effectively give us named function arguments, which makes it obvious what each one does. For an example [look here](https://github.com/Microsoft/VFSForGit/pull/908/files#diff-16401a67e162da73f89646a0727af85dR350).